app/assets/javascripts/top_menu.js

```diff
@@ -241,8 +241,39 @@
       top_menus.push(new_menu);
     });
   };
+
+  // expands .search_btn inside the given wrapper
+  // to an input .search_field with the button appended to the right.
+  $.fn.expandableSearch = function () {
+    var wrapper = $(this)
+      , form = wrapper.closest('form')
+      , btn = wrapper.find('.search_btn')
+      , input = wrapper.find('.search_field')
+      , evtns = 'click.expandableSearch';
```

Comment:
To start with some nitpicking: Did you check that style with the rest of the code? Not that our codebase is particulary homogenous, but I think we usually pute the `,` onto the upper line, without a space before it...

Still this is written down nowhere I know...